### ISSUE TITLE
React-Select v5 Bug of 2 elements in SingleValue

### ISSUE THREAD

**Issue by BarishSarac** [JedWatson/react-select#4837]
Hello,

This is to report a specific bug only exists in v5. Same code shows differently between these two versions. No matter I do in terms of CSS, I cannot fix it. I currently reverted back to v4.3.1 so don't have as much concern but unless this bug is fixed, React-Select will be a package that I and many other experiencing same bug cannot upgrade.
![v5](https://user-images.githubusercontent.com/2792881/135909366-e5577b9a-78ee-4ded-bf3c-d7c68e3ae25f.PNG)
![v4 3 1](https://user-images.githubusercontent.com/2792881/135909367-e07bcdf6-8a26-4a2c-95bb-b885f57688e1.PNG)

 ```
 SingleValue = (props) => {
    return (
      <>
        <IconButton
          onClick={this.favoritePressed}
        >
          <FavoriteFilled />
        </IconButton>
        <input
          type="text"
        />
      </>
    );
  };
```

In a nutshell, I have icon for favoriting items listed in input list so user can favorite them. No matter what I do, second item go below first item. I wrap them with DIV and gave whole bunch of CSS properties still no help.

**Comment by ebonow** [comment]
Greetings @BarishSarac,

There have been changes to the CSS styling of the ValueContainer to remove the absolute positioning and replace it with css grid styles (specifically with  `grid-area: 1/1/2/3` so that the input is given the same cell to overlap the components).

Since the SingleValue component has the styling, perhaps try and wrap your component in a div or even with the SingleValue component as a container instead of a fragment.

```jsx
SingleValue = props => (
  <components.SingleValue>
    <IconButton
         onClick={this.favoritePressed}
       >
         <FavoriteFilled />
       </IconButton>
       <input
         type="text"
       />
  </components.SingleValue>
);
```

Please let me know if this works for you.

**Comment by ebonow** [comment]
Marking this as resolved, but again please feel free to ask questions if the implementation details are still unclear.

**Comment by DoneDeal0** [comment]
Hi, have you found a fix? I also have the same issue. The `SingleValue` component is not properly displayed anymore in V5. I've tried to put it in absolute position, but it hides the value!

Here is a basic example:

```js
import { useState } from "react"
import Select from "react-select"

const options = [
  {value: "chocolate", label: "Chocolate"},
  {value: "vanilla", label: "Vanilla"}
]

export default function App() {
  const [v, setV] = useState(options[0])
  return (
     <div>
       <Select
            value={v}
            options={options}
            onChange={(option)=> setV(option)}
            components={{
                       SingleValue: ({data})=> <div>hello</div>
             }}
         />
    </div>
  );
}
```

The ugly rendering:
<img width="648" alt="Screenshot 2022-10-27 at 14 36 31" src="https://user-images.githubusercontent.com/43271780/198285972-369d787e-dc60-47c3-8755-f5bebb6552df.png">

The `SingleValue`'s `hello` div is placed haphazardly.

**Comment by ckeeney** [comment]
I am having the same issue as @DoneDeal0, and even setting `isSearchable={false}` renders some extra space for the search input:

![image](https://user-images.githubusercontent.com/8799010/205403873-a8c7ee2a-d79e-49d5-b9a0-3542b98e5177.png)

![image](https://user-images.githubusercontent.com/8799010/205404029-59fd562c-e892-4479-9bd2-1eb5c09ecde8.png)

```js

                <Select
                    styles={{
                        placeholder: (provided, state) => ({
                            textTransform: 'capitalize',
                            fontWeight: 'bold',
                        }),
                        singleValue: (provided, state) => ({
                            textTransform: 'capitalize',
                            fontSize: 18,
                            fontWeight: 'bold',
                        }),
                        control: (provided, state) => ({
                            ...provided,
                            border: 0,
                            display: 'inline-flex',
                        }),
                        container: (provided, state) => ({
                            ...provided,
                            minWidth: 200,
                            display: 'inline-block',
                        }),
                        menu: (provided, state) => ({
                            ...provided,
                            zIndex: 1000,
                        }),
                    }}
                    name="race"
                    isClearable={false}
                    isSearchable={false}
                />
```
